### ISSUE TITLE
Getting object stream

### ISSUE THREAD

**Issue by akusmin** [vtfuture/SwiftClient#1]
Is it possible to bypass original swift HttpResponseMessage content stream without copy to separate memory stream?
Now it took some time and resources to copy object to separate stream.

```c#
using (var response = await _client.SendAsync(request).ConfigureAwait(false))
                    {
                        var result = GetResponse<SwiftResponse>(response);

                        if (response.IsSuccessStatusCode)
                        {
                            result.Stream = new MemoryStream();
                            await response.Content.CopyToAsync(result.Stream).ConfigureAwait(false);
                            result.Stream.Position = 0;
                        }

                        return result;
                    }
```

**Comment by cristipufu** [comment]
Hi @akusmin,

Fixed in 718e8eecbc73789d22c69613182f8b45c738c53b, now it no longer copies the stream to a separate memory stream. It's your job now to dispose it when you no longer need it, or just let the GC handle it.

Also, if you work with very large files, you should consider using the `BufferedHTTPStream`, something like:

```cs
var stream = new BufferedHTTPStream((start, end) =>
{
        var response = client.GetObjectRange(containerId, fileId, start, end).Result;

        return response.Stream;

}, () => headObject.ContentLength);

return new FileStreamResult(stream, contentType);
```

Hope i could help you, and let me know if it's ok.

**Comment by akusmin** [comment]
Dear @cristipufu 
After several tests and investigations I've found following conclusion:
1. Currently stream from a source is still copying inside of SendAsync;
2. To eliminate stream loading following solution will work:
```c#
//HttpCompletionOption.ResponseHeadersRead - tells not to load content
var response = await _client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead).ConfigureAwait(false);

                    var result = GetResponse<SwiftResponse>(response);

                    if (response.IsSuccessStatusCode)
                    {
//Gets ReadOnlyStream. Only reference to source stream.
                        result.Stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false); 
                    }
```
p.s. BufferedHTTPStream is good but it is using similar concept with copying butch of data to a MemoryStream. So even for a small chunk there is a latency.
